tests/mtest/src/main/scala/tests/ScalaVersion.scala

```diff
@@ -0,0 +1,99 @@
+package tests
+
+import scala.util.Try
+import scala.util.matching._
+
+/** Models a parsed Scala Version number.
+ *
+ * We need to skip some tests if we are not on a very particular scala
+ * version. This class enables us to discriminate on that.
+ *
+ * It extends `Ordered` so that we can do stuff like,
+ * {{{
+ * scala> ScalaVersion(2L, 12L, 10L, "") < ScalaVersion(2L, 12L, 9L, "")
+ * res0: Boolean = false
+ * }}}
+ */
+final case class ScalaVersion(
```

Comment:
This class looks like a duplicate of `BuildTool.isCompatibleVersion` https://github.com/scalameta/metals/blob/abca8703638fe2768e0058288ae0e3decbe42e07/metals/src/main/scala/scala/meta/internal/builds/BuildTool.scala#L51

We could add a new package `scala.meta.internal.semver` with a basic helper to compare version numbers. Down the road we can hopefully replace that package with a proper library implementation like Coursier

Reply:
Do you want me to move `isCompatibleVersion` from `BulidTool` into the `scala.meta.internal.semver` package you propose? Or do you want me to add a dependency on `metals` in the `mtest` project?

Reply:
Let's move the `isCompatibleVersion` method into a `internal.semver` package inside the `mtags` project so that it can be used from mtest

Reply:
Updated.